gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/watermark/StringWatermark.java

```diff
@@ -26,19 +26,22 @@
 
 import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
-import lombok.Getter;
 
 
 /**
  * String based {@link ComparableWatermark} implementation.
  */
 @AllArgsConstructor
 @EqualsAndHashCode
-public class StringWatermark implements ComparableWatermark {
+public class StringWatermark implements ComparableWatermark<String> {
 
-  @Getter
   String value;
 
+  @Override
+  public String getValue(){
```

Comment:
Why remove the getter here and add explicit method? For LongWatermark it was essential since return type needed to be object Long instead of primitive long.

Also why is this value variable not private?